src/project/mod.rs

```diff
@@ -60,33 +59,42 @@ impl Project {
         })
     }
 
-    pub fn dependencies(&self) -> anyhow::Result<HashMap<String, NamelessMatchSpec>> {
-        let deps = self
-            .doc
-            .get("dependencies")
-            .ok_or_else(|| {
-                anyhow::anyhow!("No dependencies found in {}", consts::PROJECT_MANIFEST)
-            })?
-            .as_table_like()
-            .ok_or_else(|| {
-                anyhow::anyhow!("dependencies in {} are malformed", consts::PROJECT_MANIFEST)
-            })?;
-
-        let mut result = HashMap::with_capacity(deps.len());
-        for (name, value) in deps.iter() {
-            let match_spec = value
-                .as_str()
-                .map(|str| NamelessMatchSpec::from_str(str).map_err(Into::into))
-                .unwrap_or_else(|| {
-                    Err(anyhow::anyhow!(
-                        "dependencies in {} are malformed",
-                        consts::PROJECT_MANIFEST
-                    ))
-                })?;
-            result.insert(name.to_owned(), match_spec);
-        }
+    /// Returns the dependencies of the project.
+    pub fn dependencies(
+        &self,
+        platform: Platform,
+    ) -> anyhow::Result<HashMap<String, NamelessMatchSpec>> {
+        // Get the base dependencies (defined in the `[dependencies]` section)
+        let base_dependencies = self.manifest.dependencies.iter();
+
+        // Get the platform specific dependencies in the order they were defined.
+        let platform_specific = self
+            .target_specific_metadata(platform)
+            .flat_map(|target| target.dependencies.iter());
+
+        // Combine the specs.
+        //
+        // Note that if a dependency was specified twice the platform specific one "wins".
+        Ok(base_dependencies
+            .chain(platform_specific)
+            .map(|(name, spec)| (name.clone(), spec.clone()))
+            .collect())
+    }
 
-        Ok(result)
+    /// Returns all the targets specific metadata that apply with the given context.
+    /// TODO: Add more context here?
+    /// TODO: Should we return the selector too to provide better diagnostics later?
+    pub fn target_specific_metadata(
```

Comment:
Would it be possible to check if the defined targets are also available in the `[platforms]`? As this might be confusing if you missed it.

Reply:
Since parsing is done locally (fields are unaware of each other) its kind difficult to produce better errors. We can probably do it, but Im not sure its worth the effort now.

Reply:
No anywhere is fine. Just would like to get a warning or error that I'm doing something stupid.

Reply:
What would you like this error to look like?